src/sentry/static/sentry/app/views/eventsV2/results.tsx

```diff
@@ -478,6 +500,34 @@ export const Top = styled(Layout.Main)`
   flex-grow: 0;
 `;
 
+type SavedQueryState = AsyncComponent['state'] & {
+  savedQuery: SavedQuery;
+};
+
+class SavedQueryAPI extends AsyncComponent<Props, SavedQueryState> {
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const {organization, location} = this.props;
+    if (location.query.id) {
+      return [
+        [
+          'savedQuery',
+          `/organizations/${organization.slug}/discover/saved/${location.query.id}/`,
+        ],
+      ];
+    }
+    return [];
+  }
+
+  renderLoading() {
+    return this.renderBody();
+  }
+
+  renderBody(): React.ReactNode {
+    const {savedQuery} = this.state;
+    return <Results {...this.props} savedQuery={savedQuery} />;
```

Comment:
```suggestion
    return <Results {...this.props} savedQuery={savedQuery ?? undefined} />;
```

Reply:
When using `AsyncComponent`, `savedQuery` in `this.state` appears to be `null`. Coercing it to `undefined` to respect the prop types